api/envoy/extensions/filters/http/grpc_stats/v3/config.proto

```diff
@@ -70,6 +70,10 @@ message FilterConfig {
   // This config can fix incorrect gRPC metrics with dots because the existing stats tag extractor
   // assumes no dots in the gRPC service name. By default this is set as false.
   bool replace_dots_in_grpc_service_name = 5;
+
+  // If true, the filter will instrument `Buf Connect <https://connect.build/>` protocol messages in
+  // addition to gRPC.
+  bool enable_buf_connect_support = 6;
```

Comment:
I might be missing something here, but the filter is for GrpcStats, and Buf-Connect is a different protocol, so shouldn't there be a BufConnectStats filter (that has the relevant config)?

Reply:
What we wound up doing was merging Buf Connect support into the gRPC stats protocol due to the overlap between the protocols. (Connect is semantically compatible with gRPC, so it's somewhat akin to say, gRPC-web.) I'm open to trying to make this make more sense. Context: https://github.com/envoyproxy/envoy/pull/24836#discussion_r1065012094

Reply:
Given the similarity of the protocols I think it's reasonable to have it be part of this filter. We can potentially make this more clear in the docs. I will take a look.

Reply:
Is there any reason to actually have this be configurable? It's turned on by content-type, right? Can we just turn it on automatically and then just document that it is supported somewhere in the top level filter documentation?

Reply:
Done.